lib/Routing/Dynamic/DataObjectRouteHandler.php

```diff
@@ -98,13 +97,15 @@ public function matchRequest(RouteCollection $collection, DynamicRequestContext
      */
     private function buildRouteForFromSlug(DataObject\Data\UrlSlug $slug, DataObject\Concrete $object): DataObjectRoute
     {
+        $site = $this->siteResolver->getSite();
         $route = new DataObjectRoute($slug->getSlug());
         $route->setOption('utf8', true);
         $route->setObject($object);
         $route->setSlug($slug);
         $route->setDefault('_controller', $slug->getAction());
         $route->setDefault('object', $object);
         $route->setDefault('urlSlug', $slug);
+        $route->setDefault('site', $site ? $site->getId() : 0);
```

Comment:
I think we can just set the site object here, instead of the id (similar to the `object` above), or was there a particular reason for using the id?